src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1203,11 +1234,11 @@ public function getMethod(): string
 
         $method = strtoupper($method);
 
-        if (\in_array($method, ['GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'CONNECT', 'OPTIONS', 'PATCH', 'PURGE', 'TRACE', 'QUERY'], true)) {
```

Comment:
shouldn't we keep the default list of restricted methods, to avoid allowing (client side code) to emulate a `FOOBAR` method in the default setup when you don't configure a list ?

Reply:
it's not a restriction but a fast-path - the code I'm submitting doesn't change the behavior, it just replace a fast path + a slow path by a single fast-enough path.